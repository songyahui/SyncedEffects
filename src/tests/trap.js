"use hopscript"

var hh = require("hiphop");

var prg =
 <hh.module A B C>
  <hh.sequence>
    <hh.emit A/>
    <hh.trap T>
      <hh.sequence>
	<hh.exit T/>
	<hh.emit B/>
      </hh.sequence>
    </hh.trap>
    <hh.emit C/>
  </hh.sequence>
</hh.module>;

exports.prg = new hh.ReactiveMachine(prg, "trapsimple");
