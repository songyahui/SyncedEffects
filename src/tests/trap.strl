module trapsimple:

output A, B, C;

emit A;
trap T in
     exit T;
     emit B;
end trap;
emit C;

end module