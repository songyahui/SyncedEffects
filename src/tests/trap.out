trapsimple> ;
--- Output: A C
trapsimple> ;
--- Output:
